index.bs

```diff
@@ -1736,240 +1736,210 @@ whether they are exceptions, promise rejection values, or properties.
-    API to the web platform is generally discouraged, and should be
-    provided as a promise based API whenever possible.
+ : Design based on functionality, not based on the existing API
+ :: You will probably not be able to directly translate
+    an API available to native applications
+    to be a web API.
 
-    Doing this also allows putting the API behind a permission if it is
-    necessary, especially if the API depends on I/O or in the future may need to
-    be isolated to a separate process.
+    Instead, consider the functionality available from the native API,
+    and the user needs it addresses,
+    and design an API which meets those user needs,
+    even if the implmentation depends on the existing native API.
 
- : Don't overconstrain or overfit the API to the underlying (wrapped) OS
- :: When designing a wrapper API, you should consider how different implementations
-    (operating systems and platforms) provide this functionality. Ideally the entire
-    specification is implementable across platforms, but in some cases it may be
-    desirable to expose distinctions made on only some platforms.
+    Be particularly careful about exposing
+    the exact lifecycle and data structures of the underlying native APIs.
 
-    These distinctions should be clearly noted if they are known in advance.
+ : Make sure the web API can be implemented on more than one platform
+ :: When designing a wrapper API,
+    consider how different platforms provide its functionality.
 
- : Do not propose black-box proprietary library or dedicated hardware processor dependencies
- :: Black-box dependencies are strongly discouraged, as this not only prevents wide
-    implementor adoption, but is unhealthy to the open nature of the web.
+    Ideally, all implementations should work exactly the same,
+    but in some cases you may have a reason to expose options
+    which only work on some platforms.
+    If this happens, be sure to explain
+    how authors should write code which works on all platforms.
+    See [[#feature-detect]].
 
  : Underlying protocols should be open
- :: APIs which require exchange with external hardware or services should not depend
-    on closed protocols.
+ :: APIs which require exchange with external hardware or services
+    should not depend on closed or proprietary protocols.
+    Depending on non-open protocols undermines the open nature of the web.
 
- : Be offline friendly
- :: Web platform APIs should ideally not have a hard dependency on online services.
-    The availability of these services across different regions may vary, users
-    may be offline, and the online services have no guarantee of being always
-    available.
+ : Design APIs to handle the user being off-line
+ :: If an API depends on some service which is provided by a remote server,
+    make sure that the API functions well when the user can't access the remote server
+    for any reason.
 
  : Avoid additional fingerprinting surfaces
```

Comment:
Maybe drop this since we largely say the same above.

Reply:
(to be clear, the "Avoid additional fingerprinting surfaces" section is what I'm referring to.)